app/src/main/java/de/xikolo/extensions/NonNullMediatorLiveData.kt

```diff
@@ -26,3 +26,15 @@ fun <T> LiveData<T>.observe(owner: LifecycleOwner, observer: (t: T) -> Unit): (t
     })
     return observer
 }
+
+// observing stops when the observer function returns true
```

Comment:
What does it mean when the observer function returns true?

Reply:
The 2nd argument for `observeOnce(owner: LifecycleOwner, observer: (t: T) -> Boolean): (t: T) -> Boolean` is called `observer` and returns a Boolean. When `true` is returned, the observer is removed. 
You usually return `false` for the first observed state `NetworkCode.STARTED` and `true` for `SUCCESS`, `ERROR` etc.

Reply:
Okay.